### PR TITLE
feat: account management menu

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Add account management menu for supported multi-account wallets:

- [x] Native
- [x] Mobile
- [x] Ledger
- [x] KeepKey
- [x] Demo

Screenshot 2024-04-19 at 1 12 04 PM
Screenshot 2024-04-19 at 1 20 23 PM
Screenshot 2024-04-19 at 1 36 17 PM

Not supported:

- MetaMask
- WalletConnect (theoretically possible by mutating connection sessions, but won't do for now)
- Keplr
- Coinbase
- Xdefi

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/6716

## Risk

Small - only relates to routing and modal display

> What protocols, transaction types or contract interactions might be affected by this PR?

N/A

## Testing

Ensure that the "Manage Accounts" menu item shoes for the supported wallets above, and that clicking it opens the `AccountManagementPopover` component (unless we are connected to the demo wallet, in which case the "Connect a wallet" modal will show).

Screenshot 2024-04-19 at 10 45 39 AM

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

See testing step.

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
@woodenfurniture we can merge this if you're happy and we can wire it up to your new component when it's ready.

File: src/context/WalletProvider/DemoWallet/DemoMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const isAccountManagementEnabled = useFeatureFlag('AccountManagement')
+
+  const handleManageAccountsMenuItemClick = useCallback(() => {
+    walletDispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
+  }, [walletDispatch])
+
+  return (
+    <>
+      <MenuDivider />
+      {isAccountManagementEnabled && (
```
Comment (woodenfurniture) [inline src/context/WalletProvider/DemoWallet/DemoMenu.tsx:24]: do we still want the menu divider if account management is disabled? maybe we just reutnr null in that case?

Comment (0xApotheosis) [reply src/context/WalletProvider/DemoWallet/DemoMenu.tsx:24]: Indeed. Will switch branch context shortly and fix 🙏

Comment (0xApotheosis) [reply src/context/WalletProvider/DemoWallet/DemoMenu.tsx:23]: Addressed in https://github.com/shapeshift/web/pull/6724/commits/c4465086dcea348fd46ac6fb003fe49a6ceffb0d, good catch.

**Review by woodenfurniture — APPROVED** [review]

**Pull request merged**